### PR TITLE
Create test-package.yml

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 ## Web Server Middleware for Dart
 
-[![Build Status](https://travis-ci.org/dart-lang/shelf.svg?branch=master)](https://travis-ci.org/dart-lang/shelf)
-[![Coverage Status](https://coveralls.io/repos/dart-lang/shelf/badge.svg?branch=master)](https://coveralls.io/r/dart-lang/shelf)
```
Comment (natebosch) [inline README.md:4]: Are we losing coverage checks, or was that not working before?

Comment (kevmoo) [reply README.md:4]: It hasn't been wired up for a while!

https://coveralls.io/github/dart-lang/shelf

Jan 3, 2017!

Comment (athomas) [reply README.md:4]: FYI, dartdoc has a coveralls integration working on actions: dart-lang/dartdoc#2464.

File: .github/workflows/test-package.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  # Run on PRs and pushes to the default branch.
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: "0 0 * * 0"
+
+env:
+  PUB_ENVIRONMENT: bot.github
```
Comment (natebosch) [inline .github/workflows/test-package.yml:13]: Should we be looking at automating this in pub? We can't rely on external folks using github actions to configure this.

File: README.md (hunk 1 above)
Comment (kevmoo) [reply README.md:4]: Happy to get that going in a follow-up!

File: .github/workflows/test-package.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Check formatting
+        run: dart format --output=none --set-exit-if-changed .
+        if: always() && steps.install.outcome == 'success'
+      - name: Analyze code
+        run: dart analyze
+        if: always() && steps.install.outcome == 'success'
+
+  # Run tests on a matrix consisting of two dimensions:
+  # 1. OS: ubuntu-latest, (macos-latest, windows-latest)
+  # 2. release channel: dev, stable
```
Comment (natebosch) [inline .github/workflows/test-package.yml:41]: Should we remove this comment or rephrase to indicate that this is only 1 variant today?

Is there any particular reason to limit to ubuntu for now?

File: .github/workflows/test-package.yml (hunk 2 above)
Comment (kevmoo) [reply .github/workflows/test-package.yml:13]: Yes!

I tried to get something in the pub client – but I got pushback - https://github.com/dart-lang/pub/pull/2759

File: .github/workflows/test-package.yml (hunk 3 above)
Comment (kevmoo) [reply .github/workflows/test-package.yml:41]: done

**Review by natebosch — APPROVED** [review]

File: .github/workflows/test-package.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+env:
+  PUB_ENVIRONMENT: bot.github
+
+jobs:
+  # Check code formatting and static analysis on a single OS (linux)
+  # against Dart dev and stable.
+  analyze:
+    runs-on: ubuntu-latest
+    strategy:
```
Comment (athomas) [inline .github/workflows/test-package.yml:20]: perhaps add `fail-fast: false` (won't matter until stable is added to the matrix but easily forgotten)

**Pull request merged**